docs/how-to/crafting/create-a-component.rst

```diff
@@ -88,6 +88,6 @@ and the component file for the ``translations`` component.
    snapcraft upload hello-components_1.0_amd64.snap \
      --component translations=hello-components+translations_1.0.comp
 
-The store expects a snap and its components to be uploaded together.
-
-The component has to be uploaded alongside the snap.
+The store expects a snap and its components to be uploaded together. However, the component
+may have a different version than the snap. To dynamically set the component version,
+see :ref:`how-to-access-project-variables-across-parts-and-components`.
```

Comment:
Originally I planned to add a new section here, but it would be a perfect duplicate of what @steinbro added, so I decided to link to it instead.